api/src/routes/objects.rs

```diff
@@ -1,52 +1,71 @@
 use crate::models::response;
 use actix_web::{HttpResponse, get, web};
+use base64::prelude::*;
 use futures::TryStreamExt;
 use mongodb::{
     Collection, Database,
-    bson::{Document, doc},
+    bson::{Document, doc, document::ValueAccessResult},
 };
 use utoipa::ToSchema;
 
 #[derive(Debug, Clone, serde::Serialize, serde::Deserialize, ToSchema)]
-struct MostRecentAlert {
+struct Obj {
     object_id: String,
-    alert_metadata: serde_json::Value,
+    candidate: serde_json::Value,
     cutout_science: serde_json::Value,
     cutout_template: serde_json::Value,
     cutout_difference: serde_json::Value,
     prv_candidates: Vec<serde_json::Value>,
+    prv_nondetections: Vec<serde_json::Value>,
+    fp_hists: Vec<serde_json::Value>,
+    classifications: serde_json::Value,
     cross_matches: serde_json::Value,
+    aliases: serde_json::Value,
 }
 
 #[derive(Debug, Clone, serde::Serialize, serde::Deserialize, ToSchema)]
-struct MostRecentAlertResponse {
+struct ObjResponse {
     status: String,
     message: String,
-    data: MostRecentAlert,
+    data: Obj,
 }
 
-/// Fetch the most recent alert for a given object in a specified survey
+fn bson_docs_to_json_values(
+    array: ValueAccessResult<&Vec<mongodb::bson::Bson>>,
+) -> Vec<serde_json::Value> {
+    array
+        .unwrap_or(&vec![])
+        .iter()
+        .map(|doc| serde_json::json!(doc.clone()))
+        .collect()
+}
+
+/// Fetch an object from a given survey's alert stream
+/// Ultimately, this endpoint should format the object nicely, in a way that is useful
+/// in a way that is useful for a frontend to display object-level information.
```

Comment:
This one confused me a little. Is it fetching an alert or fetching an object?

Reply:
objects come from an alert stream. 1 object = many alerts (candidates) for it. The idea of that endpoint is for a given object ID to find the latest alert to return its metadata (candidate) + classifications + images (these come from the alert and cutout tables), and to that add the full lightcurve(s) + cross matches + survey aliases (these are in the object table).

We just try to combine all the useful data we have about an object, and serve it to the user. Often folks want to see everything about an object and we may want to serve the data in a certain way in the future, so having a dedicated "alert stream-based" object endpoint is a nice way to server those, vs requiring the user to always write an awkward sequence of find queries or aggregation pipelines.

Reply:
It's just super nice to be able to get all the data about an object just by running a single GET call where all that is needed is the objectId.

Reply:
So would it be better to simplify this as `GET /objects/{object_id}`?

Reply:
I think we need the alert stream / survey in there, otherwise we don't know where to look for that object. It's "alert data" (as in these objects are generated by an alert stream) so the current route made sense to me, but it could also be `/objects/<survey_name>/<object_id>` too! Maybe the later is more concise and clear? (`/objects/lsst/129838126351` vs `/alerts/lsst/objects/129838126351`, or `/objects/ztf/ZTF25abcedfg` vs `/alerts/ztf/objects/ZTF25abcedfg`)